sprinter/environment.py

```diff
@@ -251,13 +251,15 @@ def validate(self):
         self.phase = PHASE.VALIDATE
         self.logger.info("Validating %s..." % self.namespace)
         self.instantiate_features()
-        context_dict = {}
+
         if self.target:
+            context_dict = {
+                'config:root_dir': self.directory.root_dir,
+                'config:node': system.NODE
+            }
             for s in self.target.formula_sections():
                 context_dict["%s:root_dir" % s] = self.directory.install_directory(s)
-                context_dict['config:root_dir'] = self.directory.root_dir
-                context_dict['config:node'] = system.NODE
-                self.target.add_additional_context(context_dict)
+            self.target.add_additional_context(context_dict)
```

Comment:
Just a minor perf tweak. We were updating values that remained unchanged through the loop.